looker/resource_looker_model_set.go

```diff
@@ -0,0 +1,131 @@
+package looker
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	sdk "github.com/looker-open-source/sdk-codegen/go/sdk/v4"
+	"github.com/resolutionlife/terraform-provider-looker/internal/conv"
+)
+
+func resourceModelSet() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manages a set of Looker models",
+		CreateContext: resourceModelSetCreate,
+		ReadContext:   resourceModelSetRead,
+		UpdateContext: resourceModelSetUpdate,
+		DeleteContext: resourceModelSetDelete,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique id of the model set",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the model set",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the model set",
+			},
+			"models": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required:    true,
+				Description: "The list of models in the model set",
+			},
+		},
+	}
+}
+
+func resourceModelSetCreate(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	models, ok := d.Get("models").(*schema.Set)
+	if !ok {
+		return diag.Errorf("models is not a set")
+	}
+
+	modelsSlice, err := conv.SchemaSetToSliceString(models)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	modelSet, err := api.CreateModelSet(sdk.WriteModelSet{
+		Name:   conv.PString(d.Get("name").(string)),
+		Models: conv.PSlices(modelsSlice),
+	}, nil)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if modelSet.Id == nil {
+		return diag.Errorf("model set %s has missing id", modelSet.Name)
```

Comment:
The `modelSet.Name` is a pointer `*string` you may want to check the pointer or change the error message. This error is very unlikely to happen which is why in some other cases, the error message doesn't have much context in it.